File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/impl/SwlBase64Obfuscator.java
package i2f.swl.impl;

import i2f.codec.str.obfuscate.Base64Obfuscator;
import i2f.swl.std.ISwlObfuscator;

/**
 * @author Ice2Faith
 * @date 2024/7/10 10:35
 * @desc
 */
public class SwlBase64Obfuscator implements ISwlObfuscator {
    @Override
    public String encode(String data) {
        return Base64Obfuscator.encode(data, true);
    }

    @Override
    public String decode(String data) {
        return Base64Obfuscator.decode(data);
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-swl/core/impl/SwlBase64Obfuscator.js
import Base64Obfuscator from "../../lib/Base64Obfuscator";

/**
 * @implements {ISwlObfuscator}
 * @return {SwlBase64Obfuscator}
 */
function SwlBase64Obfuscator() {
    return {
        /**
         *
         * @param data {String}
         * @return {String}
         */
        encode(data) {
            return Base64Obfuscator.encode(data, true)
        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        decode(data) {
            return Base64Obfuscator.decode(data)
        }
    }
}

export default SwlBase64Obfuscator
